Formula/n/nghttp2.rb

```diff
@@ -34,17 +34,25 @@ class Nghttp2 < Formula
   uses_from_macos "libxml2"
   uses_from_macos "zlib"
 
-  # Fix: shrpx_api_downstream_connection.cc:57:3: error:
-  # array must be initialized with a brace-enclosed initializer
-  # https://github.com/nghttp2/nghttp2/pull/1269
-  patch do
-    on_linux do
-      url "https://github.com/nghttp2/nghttp2/commit/829258e7038fe7eff849677f1ccaeca3e704eb67.patch?full_index=1"
-      sha256 "c4bcf5cf73d5305fc479206676027533bb06d4ff2840eb672f6265ba3239031e"
-    end
+  on_macos do
+    depends_on "llvm" => :build if DevelopmentTools.clang_build_version <= 1400
+  end
+
+  on_linux do
+    fails_with gcc: "5"
+    depends_on "gcc@13"
```

Comment:
> > That's less of an issue and generally should work OK even though `libnghttp2` use an older GCC.
> 
> I'd have thought we'd be ok using a newer GCC only for `nghttp2` and not `libnghttp2`, but the `test` block disagrees, unfortunately.

Can try adding `Formula["gcc@13"].opt_lib/"gcc/13"` to `RPATH`. Not sure if it will precede `#{HOMEBREW_PREFIX}/lib/gcc/current` (otherwise may have some extra missing linkage reports whenever brew `gcc` is installed on CI).

---

EDIT: Though we may want to wait for upstream response if they will support GCC 11 given it is commonly used in LTS distros.

Reply:
> Adding a dependency with `libnghttp2` would have massive effects on the dependency tree of homebrew-core as a whole so wouldn't really be acceptable without discussion.

Is this just on Linux?

> If `libnghttp2` is broken on older Clang, we must drop support for macOS 10.11 in Homebrew/brew.

Why is that, can you elaborate?

Reply:
> I'd have thought we'd be ok using a newer GCC only for `nghttp2` and not `libnghttp2`, but the `test` block disagrees, unfortunately.

`libnghttp2` doesn't use C++ at all.

> Why is that, can you elaborate?

Because it's a dependency for curl, and we require brewed curl on older macOS.

Reply:
Actually, compiling `nghttp2` (and not `libnghttp2`) with Homebrew `gcc` should just work. It'll fail the Linux-only GCC dependency audit, though.

Reply:
> Because it's a dependency for curl, and we require brewed curl on older macOS.

Gotcha, thanks. Yeh, I think it'd either be: drop it as a dependency, drop it as a dependency on old macOS versions (dangerous with no CI coverage) or drop those macOS versions.

> It'll fail the Linux-only GCC dependency audit, though.

Seems like a good case to opt-out of.

Reply:
Let's see if using `gcc` does indeed work before we add an audit opt-out... dbded26c93988ede0fd5be08d4430babba95f34e

Reply:
Ok, that works. Homebrew/brew#17384